README.md

```diff
@@ -143,6 +143,39 @@ the mock key-value store endpoint).
 
 Notice the port number appears in three places in the `yml` for the docker-compose file.
 
+## Integration Tests and CI/CD
+
+To successfully execute integration tests against the mock server in a CI/CD context, you must ensure that the `docker-compose up` command does not return before your mock microservice container is fully instantiated and ready to receive requests.
+
+First, add a healthcheck endpoint to your OpenAPI yml file:
+```yml
+  /:
+    get:
+      summary: Check the health of the mock microservice container
+      responses:
+        200:
+          description: A successful response
+          content:
+            text/plain:
+              example: healthcheck-success
+```
+
+Then, in `docker-compose.yml`, add a `healthcheck` configuration section to your service definition. Be sure to modify the port to match the port exposed by your mock microservice container.
+```yml
+    healthcheck:
+      test: ["CMD-SHELL", "curl -f http://localhost:4200 || exit 1"]
+      interval: 10s
```

Comment:
Does this mean it waits 10 seconds between curl attempts?

Reply:
Yep, it means that the health check will first run 10s after the container is started, then again 10s after each previous check completes. According to the compose file spec, the syntax for the healthcheck block is the same as for the dockerfile HEALTHCHECK instruction.
https://github.com/compose-spec/compose-spec/blob/master/spec.md#healthcheck